pkgs/development/ocaml-modules/sha/default.nix

```
@@ -0,0 +1,20 @@
+{ lib
+, fetchgit
+, buildDunePackage
+, dune-configurator
+}:
+
+buildDunePackage rec {
+  pname = "sha";
+  version = "1.13";
+
+  useDune2 = true;
+
+  src = fetchgit {
+    url = "https://github.com/arthurteisseire/ocaml-${pname}.git";
+    sha256 = "1nfirappw3kspf45vp6f78lg36ksk30d7b2c7ymzqw71hybazs26";
+    rev = "355de5cba805b2d79e51a2636c2056e72f3f1136";
+  };
```

Why are you using your fork instead of the upstream repository? If any patches need to be added, please use the upstream repository and add the necessary patches via `patches` and `fetchpatch` (and ideally PR your changes to upstream).

Also it is best to use the release tarball as it has some preprocessing done like setting version numbers in appropriate places. 

```suggestion
  src = fetchurl {
    url = "https://github.com/djs55/ocaml-${pname}/releases/download/v${version}/${pname}-v${version}.tbz";
    sha256 = "00z2s4fsv9i1h09rj5dy3nd9hhcn79b75sn2ljj5wihlf4y4g304";
  };
```

---

I didn't managed to make it working without a patch when I was doing my PR but now it seems to work. Don't know why but the less there is, the better it is :)